Yaakov Selkowitz:
EU::Liblist::Kid refuses any linker flags which resemble $thislib.$so.
This is required by the Gtk2-Perl modules to link one against another OOTB.

I discussed this recently on gtk-perl-list.  The relevant thread:

http://mail.gnome.org/archives/gtk-perl-list/2008-March/msg00097.html
http://mail.gnome.org/archives/gtk-perl-list/2008-April/msg00000.html 

diff -u  perl-current/lib/ExtUtils/Liblist/Kid.pm.orig
--- perl-current/lib/ExtUtils/Liblist/Kid.pm.orig	2008-03-26 16:42:09.000000000 +0000
+++ perl-current/lib/ExtUtils/Liblist/Kid.pm	2008-07-03 15:57:38.265625000 +0000
@@ -132,6 +132,7 @@
 	    } elsif (-f ($fullname="$thispth/lib$thislib$Config_libext")){
 	    } elsif (-f ($fullname="$thispth/$thislib$Config_libext")){
             } elsif (-f ($fullname="$thispth/lib$thislib.dll$Config_libext")){
+            } elsif (-f ($fullname="$thispth/$thislib.dll")){
 	    } elsif (-f ($fullname="$thispth/Slib$thislib$Config_libext")){
 	    } elsif ($^O eq 'dgux'
 		 && -l ($fullname="$thispth/lib$thislib$Config_libext")
